errors.go

```diff
@@ -86,3 +89,44 @@ func NewTimeError(err error) timeError {
 		error: err,
 	}
 }
+
+// PrefixError prefixes a known err errors with prefixes from ErrorPrefixes,
+// depending on the type of err. It then adds text.
+//
+// Supported error to prefix maps are:
+//   - *oauth2.RetrieveError maps to ErrInvalidCredentials
+//
+// A classic example would look like this:
+//
+//	func Prefix() {
+// 		err := PrefixError("failed to get access token", &oauth2.RetrieveError{})
+// 		fmt.Printf("%v", err) // "ErrInvalidCredentials: failed to get access token: %w"
+//	}
+func PrefixError(text string, err error) error {
+	var prefix string
+	var e error
+
+	switch err.(type) {
+	case *oauth2.RetrieveError:
+		prefix = ErrorPrefixes[ErrInvalidCredentials]
```

Comment:
Sorry I should've mentioned earlier but based on code `oauth2.RetrieveError` is returned on _any_ non-200 http response. I was hoping that this can check 401 and return `ErrInvalidCredentials`. Otherwise, it will return `ErrInvalidCredentials` even if auth server is down and returns (say) 502.